nautobot/extras/models/tags.py

```diff
@@ -86,3 +80,8 @@ class TaggedItem(BaseModel, GenericUUIDTaggedItemBase):
 
     class Meta:
         index_together = ("content_type", "object_id")
+        # TODO: unique_together = [["content_type", "object_id", "tag"]]
```

Comment:
Not in this PR, but IMO we should do this as a part of v2 seeing as this [exact constraint is implemented in the core `TaggedItem` from `django-taggit`](https://github.com/jazzband/django-taggit/blob/master/taggit/models.py#L191).

Reply:
Opened #3437.